frontend/tests/e2e_tests/integration/05-deviceDetails.spec.ts

```diff
@@ -66,6 +66,7 @@ test.describe('Device details', () => {
     const nameInput = await page.getByLabel(/value/i);
     await nameInput.fill(demoDeviceName);
     await page.waitForTimeout(timeouts.oneSecond);
+    await page.getByRole('button', { name: /Add a rule/i }).waitFor();
```

Comment:
```suggestion
    await page.getByRole('button', { name: /Add a rule/i }).waitFor();
```
As the waiting for the button should be sufficient now - although the one extra second won't matter much in these tests.